transformer_engine/common/rmsnorm/rmsnorm_api.cpp

```diff
@@ -0,0 +1,307 @@
+/*************************************************************************
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#include <numeric>
+#include <vector>
+#include "../common.h"
+#include "rmsnorm.h"
+#include "transformer_engine/rmsnorm.h"
+
+/*
+
+Supported Type combinations:
+
+input    compute   weights   output
+=======================================
+fp32     fp32      fp32      fp32
+fp16     fp32      fp16      fp16
+bf16     fp32      bf16      bf16
+fp32     fp32      fp16      fp16
+fp32     fp32      bf16      bf16
+bf16     fp32      bf16      fp8
```

Comment:
```suggestion
input    compute   weights   output
=======================================
fp32     fp32      fp32      fp32
fp16     fp32      fp16      fp16
bf16     fp32      bf16      bf16
fp32     fp32      fp32      fp16
fp32     fp32      fp32      bf16
fp32     fp32      fp32      fp8
fp16     fp32      fp16      fp8
bf16     fp32      bf16      fp8
```
I think the documentation is also wrong in `ln_api.cpp`: https://github.com/NVIDIA/TransformerEngine/blob/d6ff6f4db9ee30429b9b3b4b06a2a9d20778c6b1/transformer_engine/common/layer_norm/ln_api.cpp#L16

Reply:
Thanks for catching that. The remark also needs to be revised -> `Input type = Weight type`